src/app/shared/services/dynamic-data/dynamic-data.service.ts

```diff
@@ -70,13 +70,27 @@ export class DynamicDataService extends AsyncServiceBase {
         const [flow_name, row_id] = args;
         await this.update("data_list", flow_name, row_id, params);
       },
+      set_item_at_index: async ({ args, params }) => {
+        const [targetIndexString, flow_name, currentItemIndex, row_ids] = args;
+        const targetIndex = this.evaluateItemIndexString(targetIndexString, currentItemIndex);
+        const targetRowId = row_ids[targetIndex];
+        if (targetRowId) {
+          await this.update("data_list", flow_name, targetRowId, params);
+        } else {
+          console.warn(`[SET ITEM AT INDEX] - No item at index ${targetIndex}`);
+        }
+      },
       set_items: async ({ args, params }) => {
         const [flow_name, row_ids] = args;
         // Hack, no current method for bulk update so make successive (changes debounced in component)
         for (const row_id of row_ids) {
           await this.update("data_list", flow_name, row_id, params);
         }
       },
+      // "Reset" the data of a given data list by removing its dynamic user overwrites
+      reset_data: async ({ args }) => {
```

Comment:
nit(non-blocking)
It might be better fitting with other item action names to just call it reset_item to remove the data_list overrides for a specific items. Could also add reset_items for bulk